src/useRouteQuery/useRouteQuery.ts

```diff
@@ -1,90 +1,111 @@
 import { reactive, watch } from 'vue'
-import { LocationQuery, LocationQueryValue, useRoute, useRouter } from 'vue-router'
+import { LocationQuery, LocationQueryValue, RouteLocationNormalizedLoaded, Router, useRoute, useRouter } from 'vue-router'
 
-type UseRouteQuery = {
+export type UseRouteQuery = {
   query: LocationQuery,
   clear: () => void,
-  set: (key: string, value: string) => void,
+  set: (key: string, value: LocationQueryValue | LocationQueryValue[]) => void,
   get: (key: string) => LocationQueryValue | LocationQueryValue[],
   remove: (key: string) => void,
 }
 
 type QueryOperation = (query: LocationQuery) => LocationQuery
 
-export function useRouteQuery(): UseRouteQuery {
-  const route = useRoute()
-  const router = useRouter()
+function factory(): () => UseRouteQuery {
   const operations = reactive<QueryOperation[]>([])
+  let route: RouteLocationNormalizedLoaded
+  let router: Router
   let interval: ReturnType<typeof setTimeout>
 
-  const clear = (): void => {
-    router.push({ query: {} })
-  }
-
-  const set = (key: string, value: string): void => {
-    query[key] = value
-  }
-
-  const get = (key: string): LocationQueryValue | LocationQueryValue[] => {
-    return query[key]
-  }
-
-  const remove = (key: string): void => {
-    delete query[key]
-  }
-
-  const query: LocationQuery = new Proxy({}, {
-    get(target, property) {
-      if (typeof property === 'symbol') {
-        return Reflect.get(target, property)
-      }
-
-      const query = applyQueryOperations(route.query, operations)
-
-      return query[property]
-    },
-    set(target, property, value) {
-      if (typeof property === 'symbol' || typeof value !== 'string') {
-        return Reflect.set(target, property, value)
-      }
-
-      const operation = updateOperationFactory(property, value)
-
-      operations.push(operation)
-
-      return true
-    },
-    deleteProperty(target, property) {
-      if (typeof property === 'symbol') {
-        return Reflect.deleteProperty(target, property)
-      }
-
-      const operation = deleteOperationFactory(property)
-
-      operations.push(operation)
-
-      return true
-    },
-  })
-
   const update = (): void => {
     const query = applyQueryOperations(route.query, operations)
 
+    operations.splice(0)
+
```

Comment:
This was a bugfix. Clear the operations array when the route is updated.

Reply:
Looking at this questions elsewhere and it seems to be, oof. Ignore me!

Reply:
Reassignment won't work on anything that is `reactive`. I think setting the length would work though I don't personally see a benefit over `splice`. But if there's a reason that's better I'll test it and if it works we can do that. 

I went with `splice(0)` purely because that was one of the solutions I found in a vue github issue.

Reply:
Yeah this is probably just on me; I've always hated that `splice` operates on the array in place because it enables operations like this which don't feel entirely clear.